share/ci/pypicongpu_generator.py

```diff
@@ -127,6 +129,54 @@ def get_all_major_pypi_versions(package_name):
     return [str(v) for v in version_map.values()]
 
 
+@typeguard.typechecked
+def combine_requirements(one: pkg_resources.Requirement, two: pkg_resources.Requirement) -> pkg_resources.Requirement:
+    if one.project_name != two.project_name:
+        print(f"requirements for {one.project_name} and {two.project_name} are incomparable!")
+        exit(1)
+
+    # accumulate all specs
+    specs: list = copy.deepcopy(one.specs)
+    specs.extend(two.specs)
+
+    if len(specs) != 0:
```

Comment:
What is this if else statement doing?

Reply:
added documentation, to make it clearer what happens